### PR TITLE
feat : Added filter context 

### PR DESCRIPTION
**Pull request by TusharJade**
Added filter context with all sorting and filtering , product listing page is done

### CODE REVIEW

File: src/components/ProductCard/ProductCard.css [hunk 1]
```diff
@@ -1,22 +1,25 @@
 .heart{
-    color: gainsboro;
+    color: rgb(208, 208, 208);
```
Comment (sumitch29) [inline src/components/ProductCard/ProductCard.css:2]: try to use color variable

File: src/pages/ProductListing/ProductListing.jsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import "./ProductListing.css";
 import { FilterBar } from "../../components/FilterBar/FilterBar";
 import { ProductsByFilter } from "../../components/ProductsByFilter/ProductsByFilter";
+
```
Comment (sumitch29) [inline src/pages/ProductListing/ProductListing.jsx:4]: Remove this

File: src/components/FilterBar/FilterBar.jsx [hunk 3]
```diff
@@ -1,96 +1,218 @@
...
+          </span>
         </div>
 
-        <div class="main-category">Sort By</div>
-        <li class="category-list">
-          <label class="category-name">
-            <input type="radio" name="sort" id="" /> Price - Low to High
+        <div className="main-category">Sort By</div>
+        <li className="category-list">
+          <label className="category-name">
```
Comment (Avin008) [inline src/components/FilterBar/FilterBar.jsx:20]: refer to the element for which this label is being used

**Review by sumitch29 — APPROVED** [review]
Ui looks good and the code is also clean some changes suggested, Good to merge

**Review by Avin008 — APPROVED** [review]
good to merge

File: src/components/ProductCard/ProductCard.css (hunk 1 above)
Comment (TusharJade) [reply src/components/ProductCard/ProductCard.css:2]: yes will do that now

File: src/pages/ProductListing/ProductListing.jsx (hunk 2 above)
Comment (TusharJade) [reply src/pages/ProductListing/ProductListing.jsx:4]: Actully if I remove this page will not work ist auto imported a component which I used on that page

File: src/components/FilterBar/FilterBar.jsx (hunk 3 above)
Comment (TusharJade) [reply src/components/FilterBar/FilterBar.jsx:20]: yup will fix that

**Pull request merged**